qa/integration/services/logstash_service.rb

```diff
@@ -113,21 +113,8 @@ def start_with_config_string(config)
   # Can start LS in stdin and can send messages to stdin
   # Useful to test metrics and such
   def start_with_stdin(pipeline_config = STDIN_CONFIG)
-    puts "Starting Logstash #{@logstash_bin} -e #{pipeline_config}"
-    Bundler.with_unbundled_env do
-      out = Tempfile.new("duplex")
-      out.sync = true
-      @process = build_child_process("-e", pipeline_config)
-      # pipe STDOUT and STDERR to a file
-      @process.io.stdout = @process.io.stderr = out
-      @process.duplex = true
-      @env_variables.map { |k, v|  @process.environment[k] = v} unless @env_variables.nil?
-      java_home = java.lang.System.getProperty('java.home')
-      @process.environment['LS_JAVA_HOME'] = java_home
-      @process.start
-      wait_for_logstash
-      puts "Logstash started with PID #{@process.pid}, LS_JAVA_HOME: #{java_home}" if alive?
-    end
+    spawn_logstash("-e", pipeline_config)
```

Comment:
This fixes an issue where a small subset of tests use the "wrong" java, since `spawn_logstash` accounts for that.